superset/config.py

```diff
@@ -585,6 +588,59 @@ class CeleryConfig:  # pylint: disable=too-few-public-methods
 # dictionary.
 JINJA_CONTEXT_ADDONS: Dict[str, Callable] = {}
 
+# A dictionary of macro template processors that gets merged into global
+# jinja template processors. The existing template processors get updated with this
+# dictionary, which means the existing keys get overwritten by the content of this
+# dictionary. The customized addons don't necessarily need to use jinjia templating
+# language. This allows you to define custom logic to process macro template.
+#
+# The example below configures a custom presto template processor which implements
+# its own logic of processing macro template with regex parsing. It uses $ style
+# macro instead of Jinja's {{ }} style.
+# By configuring it with CUSTOM_TEMPLATE_PROCESSOR, sql template on presto database
```

Comment:
Why the need to completely override jinja's template processor? JINJA_CONTEXT_ADDONS should be powerful enough

Reply:
`JINJA_CONTEXT_ADDONS` is to add customized macro function to plug into jinja templating, but what the new config does is to allow you to customize the way of evaluating template, i.e. parse `$DATE(0)` (dollar-sign) as a macro as opposed to `{{ DATE(0)}}` (jinja curly-braces).